### PR TITLE
Feature/242 243 party language serialization

### PR DESCRIPTION
**Pull request by chevalleyc**
## Changes

Support ENTRY attributes that were not persisted:
- language
- encoding
- subject
- provider

See DBEncodeTest.java for tests demonstrating this fix.

## Related issue

Fixes: https://github.com/ehrbase/project_management/issues/242 and https://github.com/ehrbase/project_management/issues/243

## Additional information and checks

- [ ] Pull request linked in changelog

### CODE REVIEW

**Comment by jakesmolka** [comment]
Quick note: not sure if related:

While the demo the rm_version attributes was now required by the validation, BUT not provided by the corresponding GET request. So the composition retrieved can't be POSTed in return (no round trip possible). But maybe the first input should have had the rm_version included, don't know.

```
"archetype_details" : {
[....]
    "rm_version" : "1.0.4"
```

**Comment by chevalleyc** [comment]
Added support of rm_version attribute in archetype_details.

**Comment by chevalleyc** [comment]
NB. This branch should be merged prior to https://github.com/ehrbase/ehrbase/pull/249 (but it is not essential)

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Java error with UTF-8 encoding resolved (see: https://github.com/ehrbase/ehrbase/pull/173)
 - AQL refactoring and fixes to support correct canonical json representation (see: https://github.com/ehrbase/ehrbase/pull/201)
 - fix terminal value test for non DataValue 'value' attribute (see: https://github.com/ehrbase/ehrbase/pull/189)
-- fix support of mandatory attributes in ENTRY specialization (see https://github.com/ehrbase/ehrbase/pull/247)
+- fix support of mandatory attributes in ENTRY specialization including rm_version (see https://github.com/ehrbase/ehrbase/pull/247)
```
Comment (jakesmolka) [inline CHANGELOG.md:43]: You added this to the wrong section. New entries should be added to the `Unreleased` section. 
Also there was a release and therefore new sections added in the current dev. There will be a merge conflict. (So maybe "soft-merge" the changelog beforehand, by copying the newest source from dev.)

**Comment by chevalleyc** [comment]
Migrated to https://github.com/ehrbase/openEHR_SDK/pull/53 and https://github.com/ehrbase/ehrbase/tree/feature/242_243_party_language_serialization_2

**Comment by jakesmolka** [comment]
@chevalleyc can this be closed now, because it was migrated over?

**Comment by chevalleyc** [comment]
Migrated to https://github.com/ehrbase/ehrbase/pull/275

**Pull request closed without merging**